From 0f94ac65017779c5f3c715b995f7bd13f94af667 Mon Sep 17 00:00:00 2001
From: Jingchang Lu <jingchang.lu@freescale.com>
Date: Wed, 7 May 2014 00:01:15 +0800
Subject: [PATCH 083/255] arm: dts: ls1021a-qds: add memioreg-muxed phy node

the memioreg muxed phys are controlled via the FPGA regs
attached to the IFC bus.

Signed-off-by: Jingchang Lu <jingchang.lu@freescale.com>
[Kevin: Original patch taken from
LS1021A-SDK-V1.1-ARM-SOURCE-20140815-yocto.iso]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm/boot/dts/ls1021a-qds.dts | 75 +++++++++++++++++++++++++++++++++------
 1 file changed, 65 insertions(+), 10 deletions(-)

diff --git a/arch/arm/boot/dts/ls1021a-qds.dts b/arch/arm/boot/dts/ls1021a-qds.dts
index 9f8e9f3..b091217 100644
--- a/arch/arm/boot/dts/ls1021a-qds.dts
+++ b/arch/arm/boot/dts/ls1021a-qds.dts
@@ -99,21 +99,22 @@
 
 &enet0 {
 	tbi-handle = <&tbi0>;
-	phy-handle = <&phy1c>;
+	phy-handle = <&physgmii1c>;
 	phy-connection-type = "sgmii";
+	status = "okay";
 };
 
 &enet1 {
 	tbi-handle = <&tbi0>;
-	phy-handle = <&phy1d>;
+	phy-handle = <&physgmii1d>;
 	phy-connection-type = "sgmii";
-	status = "disabled";
+	status = "okay";
 };
 
 &enet2 {
-	phy-handle = <&phy1>;
-	phy-connection-type = "rgmii-id";
-	status = "disabled";
+	phy-handle = <&phyrgmii3>;
+	phy-connection-type = "rgmii";
+	status = "okay";
 };
 
 &ftm0_1 {
@@ -197,7 +198,7 @@
 	/* NOR, NAND Flashes and CPLD on board */
 	ranges = <0x1 0x0 0x68000000 0x08000000
 		0x2 0x0 0x7e800000 0x00010000
-		0x3 0x0 0x7fb00000 0x00000020>;
+		0x3 0x0 0x7fb00000 0x00000100>;
 
 		nor@1 {
 			#address-cells = <1>;
@@ -280,13 +281,67 @@
 			};
 		};
 
-		cpld@3 {
+		fpga: board-control@3,0 {
 			#address-cells = <1>;
 			#size-cells = <1>;
-			compatible = "fsl,p1010rdb-cpld";
-			reg = <0x3 0x0 0x0000020>;
+			compatible = "fsl,fpga-qixis";
+			reg = <0x3 0x0 0x0000100>;
 			bank-width = <1>;
 			device-width = <1>;
+			ranges = <0 3 0 0x100>;
+
+			mdio-mux-emi1 {
+				compatible = "mdio-mux-mmioreg";
+				mdio-parent-bus = <&mdio0>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <0x54 1>; // BRDCFG4
+				mux-mask = <0xe0>; // EMI1[2:0]
+
+				/* Onboard PHYs */
+				ls1021amdio0: mdio@0 {
+					reg = <0>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+					phyrgmii1: ethernet-phy@1 {
+						reg = <0x1>;
+					};
+				};
+				ls1021amdio1: mdio@20 {
+					reg = <0x20>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+					phyrgmii2: ethernet-phy@2 {
+						reg = <0x2>;
+					};
+				};
+				ls1021amdio2: mdio@40 {
+					reg = <0x40>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+					phyrgmii3: ethernet-phy@3 {
+						reg = <0x3>;
+					};
+				};
+				ls1021amdio3: mdio@60 {
+					reg = <0x60>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+					physgmii1c: ethernet-phy@1c {
+						reg = <0x1c>;
+					};
+				};
+				ls1021amdio4: mdio@80 {
+					reg = <0x80>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+					physgmii1d: ethernet-phy@1d {
+						reg = <0x1d>;
+					};
+				};
+
+			};
+
 		};
 
 };
-- 
2.0.2

